cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/sampling/prometheus/model/PrometheusMetric.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2020 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License"). See License in the project root for license information.
+ */
+
+package com.linkedin.kafka.cruisecontrol.monitor.sampling.prometheus.model;
+
+import java.util.Objects;
+import javax.annotation.Nullable;
+
+import com.google.gson.annotations.SerializedName;
+
+public class PrometheusMetric {
+    @SerializedName("__name__")
+    final private String _domainName;
+    @SerializedName("instance")
+    final private String _instance;
+    @SerializedName("job")
+    final private String _job;
+    @SerializedName("name")
+    final private String _name;
+    @SerializedName("topic")
+    final @Nullable private String _topic;
+    @SerializedName("partition")
+    final @Nullable private String _partition;
+
+    public PrometheusMetric(
+        String domainName, String instance, String job, String name,
+        @Nullable String topic, @Nullable String partition) {
+        _domainName = domainName;
+        _instance = instance;
```

Comment:
We are using GSON to deserialize the JSON response returned by Prometheus API. Unfortunately, GSON does not use a non no-args constructor. It simply allocates the object, and then sets the fields using reflection (not using the setters). The only way to force GSON to use a custom constructor is to define a `JsonDeserializer` class.

Since this constructor is solely used in unit tests, I have not kept the validation here, but in the reader of `instance()`.